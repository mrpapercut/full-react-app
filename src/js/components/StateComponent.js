import {Component} from 'react';

import createElement from '~/util/createElement';

const [div, h2, label, span, input, button] = ['div', 'h2', 'label', 'span', 'input', 'button'].map(createElement);

class StateComponent extends Component {
    constructor(props) {
        super(props);

        this.state = this.getInitialState();
    }

    getInitialState() {
        return {
            checkboxChecked: false
        };
    }

    render() {
        return div({
            className: 'statecomponent'
        },
            h2({
                className: 'pagetitle'
            }, 'State Component'),
            div({},
                label({
                    htmlFor: 'checkbox'
                },
                    span({}, 'Checkbox')
                ),
                input({
                    id: 'checkbox',
                    type: 'checkbox',
                    checked: this.state.checkboxChecked,
                    onChange: e => this.setState({
                        checkboxChecked: !this.state.checkboxChecked
                    })
                })
            )
        );
    }
}

export default StateComponent;
