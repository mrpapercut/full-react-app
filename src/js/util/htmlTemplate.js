const htmlTemplate = ({title, body, scriptTags, preloadedState}) => `<!doctype html>
<html>
<head>
    <meta charset="utf8"></meta>
    <title>${title}</title>
    <base href="/">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>
    <div id="appwrapper">${body}</div>
    <script>${preloadedState}</script>
    ${scriptTags}
</body>
</html>
`;

export default htmlTemplate;
