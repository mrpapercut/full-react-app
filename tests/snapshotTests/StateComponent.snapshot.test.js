import {createElement} from 'react';
import renderer from 'react-test-renderer';

import {wrapElementWithRouter} from '../util/snapshot-test-util';

import StateComponent from '~/components/StateComponent';

test('StateComponent', () => {
    const component = renderer.create(wrapElementWithRouter(createElement(StateComponent)));
    expect(component.toJSON()).toMatchSnapshot();
});
