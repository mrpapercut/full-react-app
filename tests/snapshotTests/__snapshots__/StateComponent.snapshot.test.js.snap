// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateComponent 1`] = `
<div
  className="statecomponent"
>
  <h2
    className="pagetitle"
  >
    State Component
  </h2>
  <div>
    <label
      htmlFor="checkbox"
    >
      <span>
        Checkbox
      </span>
    </label>
    <input
      checked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
